src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignedPackageVerifierSettings.cs

```diff
@@ -54,50 +91,124 @@ public sealed class SignedPackageVerifierSettings
             AllowMultipleTimestamps = allowMultipleTimestamps;
             AllowNoTimestamp = allowNoTimestamp;
             AllowUnknownRevocation = allowUnknownRevocation;
+            AllowNoRepositoryCertificateList = AllowNoRepositoryCertificateList;
```

Comment:
Houston...

This tells me that your tests are wrong.  For example, see [here](https://github.com/NuGet/NuGet.Client/pull/2139/files#diff-116f0a8ca0f554a611d23e3b9a759121R284) and [here](https://github.com/NuGet/NuGet.Client/pull/2139/files#diff-116f0a8ca0f554a611d23e3b9a759121R295).

You should have tests for both `false` and `true`.

I sometimes write tests (like [this one](https://github.com/NuGet/NuGet.Client/blob/870340673ba75a95407ab55ce1f46f57fe9bf25b/test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/Messages/CopyNupkgFileRequestTests.cs#L72)) to ensure this kind of bug doesn't happen.